diff --git a/node_modules/@tanstack/start-plugin-core/dist/esm/vite/start-compiler-plugin/plugin.js b/node_modules/@tanstack/start-plugin-core/dist/esm/vite/start-compiler-plugin/plugin.js
index b3cf7ac..3c89b28 100644
--- a/node_modules/@tanstack/start-plugin-core/dist/esm/vite/start-compiler-plugin/plugin.js
+++ b/node_modules/@tanstack/start-plugin-core/dist/esm/vite/start-compiler-plugin/plugin.js
@@ -162,7 +162,7 @@ function startCompilerPlugin(opts) {
 							resolveId: async (source, importer) => {
 								const r = await this.resolve(source, importer);
 								if (r) {
-									if (!r.external) return cleanId(r.id);
+									if (!r.external) return r.id.startsWith("\0tanstack-start-import-protection:") ? r.id : cleanId(r.id);
 								}
 								return null;
 							}
